dl1_data_handler/dl_eventsources.py

```diff
@@ -0,0 +1,302 @@
+        self.meta = file1["RunHeaders"]
+        self._mc_header = self._parse_mc_header()
+        
+    @property
+    def is_simulation(self):
+        """
+        Whether the currently open file is simulated
+
+
+        Returns
+        -------
+        bool
+
+        """
+        return True
+
+    @property
+    def datalevels(self):
+        """
+        The datalevels provided by this event source
+
+
+        Returns
+        -------
+        tuple[str]
+
+        """
+        return ('R0','R1','DL0')
+    @property
+    def subarray(self):
+        """
+        Obtain the subarray from the EventSource
+
+
+        Returns
+        -------
+        ctapipe.instrument.SubarrayDescription
+
+        """
+        return self.magic_subarray
+
+    @property
+    def obs_id(self):
+        """
+        The current observation id
+
+
+        Returns
+        -------
+        int
+
+        """
+        return self.run_number
+
+    @staticmethod
+    def is_compatible(file_mask):
+        """
+        This method checks if the specified file mask corresponds
+        to MAGIC data files. The result will be True only if all
+        the files are of ROOT format and contain an 'Events' tree.
+        Parameters
+        ----------
+        file_mask: str
+            A file mask to check
+        Returns
+        -------
+        bool:
+            True if the masked files are MAGIC data runs, False otherwise.
+        """
+
+        is_magic_root_file = True
+
+        file_list = glob.glob(file_mask)
+
+        for file_path in file_list:
+            try:
+                import uproot
+
+                try:
+                    with uproot.open(file_path) as input_data:
+                        if 'Events' not in input_data:
+                            is_magic_root_file = False
+                except ValueError:
+                    # uproot raises ValueError if the file is not a ROOT file
+                    is_magic_root_file = False
+                    pass
+
+            except ImportError:
+                if re.match(r'.+_m\d_.+root', file_path.lower()) is None:
+                    is_magic_root_file = False
+
+        return is_magic_root_file  
+
+    def _generator(self):
+        """
+        Stereo event generator. Yields DataContainer instances, filled
+        with the read event data.
+        
+        Returns
+        -------
+        
+        """  
+        counter = 0
+        data = DataContainer()
+        data.meta['origin'] = "MAGIC"
+        data.meta['input_url'] = self.input_url
+        data.meta['is_simulation'] = True
+        data.mcheader = self._mc_header
+        #Reading data from root file for Events table
+        eventid_M1 = np.asarray(self.event_M1["MRawEvtHeader.fStereoEvtNumber"].array())
+        eventid_M2 = np.asarray(self.event_M2["MRawEvtHeader.fStereoEvtNumber"].array())
+        
+        zenith = np.asarray(self.event_M1["MSrcPosCam.fY"].array())
```

Comment:
Can we rename zenith to 'src_pos_cam_Y' or similar?

Reply:
Done